blockchains/receipts/lib/helper.js

```diff
@@ -18,49 +17,57 @@ const parseTransactionReceipts = (responses) => {
   return receipts;
 };
 
-const decodeLog = (log) => {
+const decodeLog = (log, web3Wrapper) => {
   collection.forEach(['blockNumber', 'blockHash', 'transactionHash', 'transactionIndex'],
-  key => object.unset(log, key));
+    key => object.unset(log, key));
 
   collection.forEach(['logIndex', 'transactionLogIndex'],
-    key => log[key] = Web3.utils.hexToNumber(log[key])
+    key => {
+      if (Object.prototype.hasOwnProperty.call(log, key) && log[key] !== undefined) {
+        log[key] = web3Wrapper.parseHexToNumber(log[key]);
+      }
+    }
   );
 
   return log;
 };
 
-const columnizeLogs = (logs) => {
+const columnizeLogs = (logs, web3Wrapper) => {
   if (logs.length === 0) { return []; }
 
-  const decodedLogs = collection.map(logs, decodeLog);
+  const decodedLogs = collection.map(logs, log => decodeLog(log, web3Wrapper));
   const logKeys = object.keys(decodedLogs[0]);
   const result = {};
   collection.forEach(logKeys, key => result[`logs.${key}`] = decodedLogs.map(log => log[key]));
 
   return result;
 };
 
-const decodeReceipt = (receipt) => {
-  const clonedReceipt = lang.clone(receipt);
+const decodeReceipt = (receipt, web3Wrapper) => {
+  const clonedReceipt = lang.cloneDeep(receipt);
 
   collection.forEach(['blockNumber', 'status', 'transactionIndex'],
-    key => clonedReceipt[key] = Web3.utils.hexToNumber(clonedReceipt[key])
+    key => {
+      if (Object.prototype.hasOwnProperty.call(clonedReceipt, key) && clonedReceipt[key] !== undefined) {
```

Comment:
We need to check if those fields are actually present, otherwise we would set `null`.